### PR TITLE
IT Integratord: Add new test integratord alerts messages

### LINKED ISSUE #2944: Create tests for different types of inputs for the integrator module
Version | Component | Core issue | Core development branch | QA development branch 
-- | -- | -- | --|--|
4.3  | Integrator  | https://github.com/wazuh/wazuh/issues/12664   |  4.3 

# Description

As part of the fix introduced in https://github.com/wazuh/wazuh/issues/12664, we need to create some tests to validate it.

These are some scenarios that need to be tested:

- If the alert read is valid JSON, the alert is parsed and returned.

- If the alert read is invalid JSON, an error message is displayed and NULL is returned.

> "Invalid JSON alert read from 'logs/alerts/alerts.json': 'Invalid JSON alert example'"

- If the alert read is too long (more than 64Kb), an error message is displayed and NULL is returned.

> "Overlong JSON alert read from 'logs/alerts/alerts.json'"

- If the file cannot be read, and it is none of the above cases, check if the inode changed and try again. For this case, it will retry infinitely until it can read a complete line.

> "Can't read from 'logs/alerts/alerts.json'. Trying again"

Take in mind that the input of integrator module is the `alerts.json` file.

## Configuration option
- **Local Internal Options** : `syscheck.debug=2 integrador.debug=2`
- Configure folder to monitor with syscheck on realtime
- Configure virustotal integration
   ```xml
     <integration>
       <name>virustotal</name>
       <api_key>VIRUS_TOTAL_API_KEY</api_key> 
       <rule_id>100200,100201</rule_id>
       <alert_format>json</alert_format>
     </integration>
   ```
# Use cases

- [x] Case 1 - Insert valid alert:

    - Insert a valid alert into `alerts.json`
    - Wait for virustotal response in `alerts.json`

- [x] Case 2 - Insert invalid alert:

    - Insert a invalid alert into `alerts.json` (broken json)
    - Wait for error response in `ossec.log`
        - `"Invalid JSON alert read from 'logs/alerts/alerts.json': 'Invalid JSON alert example'"`

- [x] Case 3 - Insert Overlong alert (64kb+):

    - Insert an overlong alert into `alerts.json`
    - Wait for error response in `ossec.log`
        - `"Overlong JSON alert read from 'logs/alerts/alerts.json'"`

- [x] Case 4 - alerts.json cannot be read:

    - Generate continued alerts for integrator to keep reading the alerts.json
       - Insert content continually into file in the monitored folder
    - Create copy of `alerts.json`
    - Replace `alerts.json` with copy
    - Wait for error response in `ossec.log`
        - `"Can't read from 'logs/alerts/alerts.json'. Trying again"`
        -  `"Alert file inode changed. Reloading."`
                
# Development stage

- [x] In research
- [x] In development
    - [x] Linux 
- [x] In testing phase
    - [x] Linux 
- [x] In Review
- [ ] Completed

### PR DESCRIPTION
**Pull request by Deblintrake09**
|Related issue|
|-------------|
| Closes #2944 |

## Description

This PR aims to add coverage to the Integratord module, by generating alerts and checking the responses on the ossec.log.

### Added
The following test modules were added as part of the the new `test_integratord` test suite
- test_integratord_change_inode_alert.py
- test_integratord_read_json_alerts.py
- test_integratord_read_json_file_deleted.py

---

## Testing performed
| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @Deblintrake09 (Developer)  |           | [:green_circle: ](https://ci.wazuh.info/job/Test_integration/30377/) [:green_circle:](https://ci.wazuh.info/job/Test_integration/30378/) [:green_circle:](https://ci.wazuh.info/job/Test_integration/30379/) | [:green_circle: ](https://ci.wazuh.info/view/Tests/job/Test_integration/29884) [:green_circle: ](https://ci.wazuh.info/view/Tests/job/Test_integration/29887) [:green_circle: ](https://ci.wazuh.info/view/Tests/job/Test_integration/29888) |  Centos  |  [9ee89bd](https://github.com/wazuh/wazuh-qa/pull/3125/commits/9ee89bd5d5694de84fc89482771b07a36cb7fd10)  | [package](https://packages-dev.wazuh.com/warehouse/test/4.3/rpm/var/wazuh-manager-4.3.7-2944.fix.integratord.read.alerts.x86_64.rpm) |
| @user (Reviewer)   |           | ⚫⚫⚫ | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |        |         | Nothing to highlight |

### CODE REVIEW

File: tests/integration/test_integratord/data/test_cases/cases_integratord_read_json_alerts.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+- name: 'Read valid json alert'
+  description: 'Read a valid alert from alerts.json'
+  configuration_parameters:
+    API_KEY: d7f1a3b90040e663357def13d16cf0481ebbda4869caba712172bd7cc79dbc7a
```
Comment (jmv74211) [inline tests/integration/test_integratord/data/test_cases/cases_integratord_read_json_alerts.yaml:4]: Remove API keys from the repository

File: tests/integration/test_integratord/data/test_cases/cases_integratord_read_json_alerts.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+- name: 'Read valid json alert'
```
Comment (jmv74211) [inline tests/integration/test_integratord/data/test_cases/cases_integratord_read_json_alerts.yaml:1]: Check the name convention from the other tests

File: tests/integration/test_integratord/test_integratord_read_json_alerts.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        error_message = ERR_MSG_VIRUSTOTAL_ALERT_NOT_DETECTED
+
+    elif metadata['alert_type'] == 'invalid':
+        callback = CB_INVALID_JSON_ALERT_READ
+        error_message = ERR_MSG_INVALID_ALERT_NOT_FOUND
+
+    elif metadata['alert_type'] == 'overlong':
+        callback = CB_OVERLONG_JSON_ALERT_READ
+        error_message = ERR_MSG_OVERLONG_ALERT_NOT_FOUND
+    # Add 90kb of padding to alert to make it go over the allowed value of 64KB.
```
Comment (jmv74211) [inline tests/integration/test_integratord/test_integratord_read_json_alerts.py:124]:
```suggestion
        # Add 90kb of padding to alert to make it go over the allowed value of 64KB.
```

**Review by jmv74211 — CHANGES_REQUESTED** [review]
I have only checked the first test case. Please, apply the mentioned changes, and I will review the next ones after that.

File: CHANGELOG.md [hunk 4]
```diff
@@ -16,8 +16,8 @@ Release report: TBD
 Wazuh commit: https://github.com/wazuh/wazuh/commit/be15851b8ead7512d9cd4ef1ee18b3b953173211 \
 Release report: https://github.com/wazuh/wazuh/issues/14188
 
-### Added
-
+## Added
+- Add Integratord IT - new test_integratord suite ([#3125](https://github.com/wazuh/wazuh-qa/pull/3125)) \- (Framework + Tests)
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply CHANGELOG.md]: Fixed

File: tests/integration/test_integratord/data/test_cases/cases_integratord_read_json_alerts.yaml (hunk 2 above)
Comment (Deblintrake09) [reply tests/integration/test_integratord/data/test_cases/cases_integratord_read_json_alerts.yaml]: Ithis format is the same as in the VDT test suite refactor

File: tests/integration/test_integratord/test_integratord_read_json_alerts.py (hunk 3 above)
Comment (Deblintrake09) [reply tests/integration/test_integratord/test_integratord_read_json_alerts.py:125]: Fixed

File: tests/integration/test_integratord/conftest.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+
+import pytest
+
+from wazuh_testing.tools import LOG_FILE_PATH
+from wazuh_testing.tools.monitoring import FileMonitor, callback_generator
+from wazuh_testing.modules import integratord as integrator
+
+
+# Fixtures
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply tests/integration/test_integratord/conftest.py:15]: Fixed

**Review by jmv74211 — APPROVED** [review]

**Pull request merged**